prestoadmin/prestoclient.py

```diff
@@ -126,19 +143,28 @@ def get_response_from(self, uri):
         Sends a GET request to the Presto server at the specified next_uri
         and updates the response
         """
-        try:
-            conn = urlopen(uri, None, URL_TIMEOUT_MS)
-            answer = conn.read()
-            conn.close()
+        parts = list(urlparse.urlsplit(uri))
+        parts[0] = None
+        parts[1] = None
+        location = urlparse.urlunsplit(parts)
 
-            self.response_from_server = json.loads(answer)
-            _LOGGER.info("GET request successful for uri: " + uri)
-            return True
-        except (HTTPError, URLError) as e:
-            _LOGGER.error("Error opening the presto response uri: " +
-                          str(e.reason))
+        conn = self._get_connection()
+        conn.request("GET", location)
+        response = conn.getresponse()
+
+        if response.status != 200:
+            conn.close()
+            _LOGGER.error("Connection error: " +
```

Comment:
whatever's in the "location" variable is what I was thinking. server probably would also be helpful